services/venice-controller/src/main/java/com/linkedin/venice/controller/HelixVeniceClusterResources.java

```diff
@@ -206,6 +208,23 @@ public HelixVeniceClusterResources(
     veniceAdminStats = new VeniceAdminStats(metricsRepository, "venice-admin-" + clusterName);
     this.storagePersonaRepository =
         new StoragePersonaRepository(clusterName, this.storeMetadataRepository, adapterSerializer, zkClient);
+    if (!config.isParent() && config.isSystemStoreHealthCheckEnabled()) {
+      if (!admin.getPushStatusStoreReader().isPresent()) {
+        throw new VeniceException("Da Vinci push status reader is not enabled.");
+      }
+      if (!admin.getPushStatusStoreWriter().isPresent()) {
+        throw new VeniceException("Da Vinci push status writer is not enabled.");
+      }
+      this.systemStoreHealthCheckService = new SystemStoreHealthCheckService(
+          storeMetadataRepository,
+          admin.getMetaStoreReader(),
+          admin.getMetaStoreWriter(),
+          admin.getPushStatusStoreReader().get(),
+          admin.getPushStatusStoreWriter().get(),
+          config.getDaVinciPushStatusScanIntervalInSeconds());
```

Comment:
This param doesn't look right, and it should be using the newly added interval config.